### PR TITLE
Make EnvoyProxyMiddleware a Gin-style middleware

### PR DESCRIPTION
**Pull request by nikicc**
Relates to https://viaduct-company.monday.com/boards/959188993/pulses/1182990615

### CODE REVIEW

**Review by devstein — COMMENTED** [review]
@nikicc What are your thoughts on separating `gin` utilities from the `http` utilities? Essentially creating a new folder 

```bash
ginutils/middlewares/middlewares.go
```

**Comment by nikicc** [comment]
I like it. Especially as the separation between the http vs gin approach will be more evident.

**Comment by nikicc** [comment]
@devstein please check again, I've migrated to `ginutils/middlewares/`.

File: ginutils/middlewares/middlewares.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package middlewares
+
+import (
+	"github.com/gin-gonic/gin"
+	"net/http"
+	"net/url"
```
Comment (devstein) [inline ginutils/middlewares/middlewares.go:6]: import order

**Review by devstein — APPROVED** [review]

File: ginutils/middlewares/middlewares.go (hunk 1 above)
Comment (nikicc) [reply ginutils/middlewares/middlewares.go]: 🙈  Fixed now.

**Pull request merged**